### ISSUE TITLE
if workers are ordered to build a building but the last main gets destroyed, a game over message is displayed

### ISSUE THREAD

**Issue by andy5995** [ZetaGlest/zetaglest-source#174]
Situation.. I order some workers to build a building, but in the meantime, my last main is destroyed. I then have 0 gold until my workers build the main I told them to build.

In the meantime, the game registers that it's completed, even though my workers are still building a main, and it gets completed.

**Comment by KeithSammut** [comment]
Yeah it registered that the game was won by me, but Andy had just been building a main building.
